### PR TITLE
Add legal-auto bot talking to the legaldb API

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: legal-auto.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
```
Comment (okurz) [inline legal-auto.py]: `#!/usr/bin/env python` ?

File: legal-auto.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                                     reqid = req.reqid, newstate = state,
+                                     by_group=self.review_group,
+                                     by_user=self.review_user, message=self.message)
+        self.delete_from_db(req.reqid)
+
+class CommandLineInterface(ReviewBot.CommandLineInterface):
+
+    def __init__(self, *args, **kwargs):
+        ReviewBot.CommandLineInterface.__init__(self, args, kwargs)
+        self.clazz = LegalAuto
```
Comment (okurz) [inline legal-auto.py]: clazz?

File: legal-auto.py (hunk 1 above)
Comment (coolo) [inline legal-auto.py]: no

File: legal-auto.py (hunk 2 above)
Comment (coolo) [inline legal-auto.py]: yes

**Comment by coolo** [comment]
Dare to explain why I would care about pep8 (whatever that is?) for my program?

**Comment by okurz** [comment]
http://lmgtfy.com/?q=pep8

File: legal-auto.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        if self.dryrun:
+            self.logger.debug("dry setting %s to %s with %s"%(req.reqid, state, self.message))
+            return
+
+        self.logger.debug("setting %s to %s"%(req.reqid, state))
+        osc.core.change_review_state(apiurl = self.apiurl,
+                                     reqid = req.reqid, newstate = state,
+                                     by_group=self.review_group,
+                                     by_user=self.review_user, message=self.message)
+        self.delete_from_db(req.reqid)
```
Comment (jberry-suse) [inline legal-auto.py]: This appears to be the only unique line short of dropping functionality from the original. Perhaps call super() and check `self.dryrun` before `delete_from_db()`?

Comment (jberry-suse) [inline legal-auto.py]: This appears to be the only unique line short of dropping functionality from the original. Perhaps call super() and check `self.dryrun` before `delete_from_db()`? The message should be settable via the standard `self.review_messages`.

File: legal-auto.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+    def __init__(self, *args, **kwargs):
+        ReviewBot.ReviewBot.__init__(self, *args, **kwargs)
+
+        self.do_comments = True
+        self.legaldb = None
+        self.commentapi = CommentAPI(self.apiurl)
+        self.apinick = None
+        self.message = None
+        if self.apiurl == 'https://api.suse.de':
```
Comment (jberry-suse) [inline legal-auto.py]: nit: `if self.ibs:`

File: legal-auto.py (hunk 3 above)
Comment (coolo) [inline legal-auto.py]: this is also about the message.

File: legal-auto.py (hunk 4 above)
Comment (coolo) [inline legal-auto.py]: yeah, but that is pretty non-symmetric then - as there is no self.obs

File: legal-auto.py (hunk 1 above)
Comment (mimi1vx) [inline legal-auto.py]: pretty bad idea use `env`

**Pull request merged**